src/main_menu.cpp

```diff
@@ -1006,3 +887,141 @@ bool main_menu::new_character_tab()
+                    nc_color color1, color2;
+                    if( world_name == "TUTORIAL" || world_name == "DEFENSE" ) {
+                        color1 = c_ltcyan;
+                        color2 = h_ltcyan;
+                    } else {
+                        if( world_generator->world_need_lua_build( world_name ) ) {
+                            color1 = c_dkgray;
+                            color2 = h_dkgray;
+                        } else {
+                            color1 = c_white;
+                            color2 = h_white;
+                        }
+                    }
+                    mvwprintz( w_open, line, 15 + iMenuOffsetX + extra_w / 2,
+                               ( sel2 == i ? color2 : color1 ), "%s (%d)",
+                               world_name.c_str(), savegames_count );
+                }
+            }
+            wrefresh( w_open );
+            refresh();
+            const std::string action = ctxt.handle_input();
+            if( world_generator->all_worldnames.empty() && ( action == "DOWN" || action == "CONFIRM" ) ) {
+                layer = 1;
+            } else if( action == "DOWN" ) {
+                if( sel2 > 0 ) {
+                    sel2--;
+                } else {
+                    sel2 = world_generator->all_worldnames.size() - 1;
+                }
+            } else if( action == "UP" ) {
+                if( sel2 < ( int )world_generator->all_worldnames.size() - 1 ) {
+                    sel2++;
+                } else {
+                    sel2 = 0;
+                }
+            } else if( action == "LEFT" || action == "QUIT" ) {
+                layer = 1;
+            } else if( action == "RIGHT" || action == "CONFIRM" ) {
+                if( sel2 >= 0 && sel2 < ( int )world_generator->all_worldnames.size() ) {
+                    layer = 3;
+                    sel3 = 0;
+                }
+            }
+        } else if( layer == 3 && sel1 == 2 ) {
+            bool available = false;
+
+            savegames = world_generator->all_worlds[world_generator->all_worldnames[sel2]]->world_saves;
+            std::string wn = world_generator->all_worldnames[sel2];
+
+            //hide savegames if lua is not available for a lua-built world
+            if( ( wn != "TUTORIAL" && wn != "DEFENSE" ) && world_generator->world_need_lua_build( wn ) ) {
+                savegames.clear();
+                mvwprintz( w_open, iMenuOffsetY - 2, 15 + iMenuOffsetX + extra_w / 2,
+                           c_red, _( "This world requires the game to be compiled with Lua." ) );
+            } else if( savegames.empty() ) {
+                mvwprintz( w_open, iMenuOffsetY - 2, 19 + 19 + iMenuOffsetX + extra_w / 2,
+                           c_red, _( "No save games found!" ) );
+            } else {
+                for( std::vector<std::string>::iterator it = savegames.begin();
```

Comment:
Except there a couple places within the loop that use `it` as opposed to `*it`. But most of that code can definitely be simplified by using STL functions.